src/all/yellownote/src/eu/kanade/tachiyomi/extension/all/yellownote/YellowNotePreferences.kt

```diff
@@ -0,0 +1,42 @@
+package eu.kanade.tachiyomi.extension.all.yellownote
+
+import android.content.Context
+import android.content.SharedPreferences
+import androidx.preference.EditTextPreference
+import eu.kanade.tachiyomi.lib.i18n.Intl
+import okhttp3.HttpUrl.Companion.toHttpUrl
+
+object YellowNotePreferences {
+
+    private const val PS_KEY_ROOT = "XChina"
+    private const val PS_KEY_DOMAIN = "$PS_KEY_ROOT::DOMAIN"
+
+    private const val DEFAULT_BASE_URL = "https://xchina.co"
+
+    internal fun SharedPreferences.baseUrl(subdomain: String?): String {
```

Comment:
is a baseurl pref needed?
also are the mirrors known? if so make a mirror pref, without option to enter custom url

Reply:
then please implement mirror pref in [such a way](https://github.com/keiyoushi/extensions-source/blob/71d5f45eacbf5e41db821f08af1df638ecd8c5a6/src/ar/teamx/src/eu/kanade/tachiyomi/extension/ar/teamx/TeamX.kt#L230-L261)

basically save current mirror from code in another pref value, and compare that to one in code, if they are different (ext updated to change it), change both in pref to match it